shell/artifacts/Social/source/EditPost.js

```diff
@@ -54,70 +54,46 @@ defineParticle(({DomParticle, log}) => {
+  <textarea value="{{message}}" on-input="_onTextInput"></textarea>
 </div>
   `.trim();
 
   return class extends DomParticle {
     get template() {
       return template;
     }
-    _getInitialState() {
-      return {message: ''};
-    }
-    _willReceiveProps(props) {
-      if (props.user) {
-        this._setState({user: props.user});
-      }
-    }
-    _update(props, state, lastProps, lastState) {
-      const {post} = props;
-      // TODO(wkorman): We have to use update here because we need lastProps
-      // which isn't currently available in _render. The overall hijinks in
-      // which we engage below are in order to make sure we only update the
-      // textarea text when a new message is supplied by WritePosts. An
-      // alternate implementation could be to scrutinize the message id and if
-      // it changes we know it's time to start over and so we clear everything
-      // out.
-      const lastMessage = lastProps.post ? lastProps.post.message : null;
-      state.textMessage = null;
-      if (post && post.message !== lastMessage) {
-        state.textMessage = post.message;
-        this._updateSaveButton(post.message);
-      }
-      super._update(props, state, lastProps, lastState);
+    willReceiveProps({post}, state) {
+      state.message = post && post.message;
     }
-    _render({post}, state) {
-      let saveClasses = ['material-icons'];
-      if (state.saveButtonActive) {
-        saveClasses.push('button-live');
-      }
-      saveClasses = saveClasses.join(' ');
-      const model = {saveClasses};
-      if (state.textMessage !== null) {
-        model.message = state.textMessage;
+    render({user, post}, {message, savePost}) {
+      if (savePost) {
+        this.savePost(user, message);
       }
+      const saveButtonActive = message && (message.trim().length > 0);
+      const model = {
+        saveClasses: saveButtonActive ? 'material-icons button-live' : 'material-icons',
+        message
+      };
       return model;
     }
-    _updateSaveButton(message) {
-      this._setState({saveButtonActive: message.trim().length > 0});
-    }
-    _onUpdatePostButtonState(e, state) {
-      this._updateSaveButton(e.data.value);
+    setHandle(name, data) {
+      const handle = this._views.get(name);
+      handle.set(new (handle.entityClass)(data));
     }
-    // TODO(wkorman): Add onDeletePost, onAttachPost.
-    _onSavePost(e, state) {
-      const Post = this._views.get('posts').entityClass;
-      this._views.get('post').set(new Post({
-        message: state.message,
+    savePost(user, message) {
+      this.setHandle('post', {
+        message: message,
         createdTimestamp: Date.now(),
-        author: state.user.id
-      }));
+        author: user.id
+      });
+      this._setState({savePost: false});
     }
-    _onCaptureText(e, state) {
-      if (state.message !== e.data.value) {
-        this._setState({message: e.data.value});
-      }
+    _onTextInput(e) {
```

Comment:
Checking thought on naming -- so keep `_` prefix on event handlers but omit from essentially Particle-private methods like `savePost` and `setHandle`? I was expecting consistency either direction, am fine with it as is, but I wanted to understand the thought since method naming was one of the specific items targeted by this PR.

Reply:
There's no reason to have underscores on event handlers. It escaped my notice because these are not part of Xen or DomParticle proper, but are the domain of individual Particles.